modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/ProjectRepository.java

```diff
@@ -159,9 +157,13 @@ else if (field.equals(DBConstants.Project.DESCRIPTION)) {
 
         }
 
-        if (!accessibleProjectIds.isEmpty() && accessibleProjectIds != null) {
+        if (accessibleProjectIds != null && !accessibleProjectIds.isEmpty()) {
             logger.debug("Filter Projects by Accessible Project IDs");
-            query += " P.projectId IN :" + accessibleProjectIds;
+            query += "P.projectId IN (";
```

Comment:
@tilaks26 The error is because there are these hyphens in the query variable name (`:projectName_a7220549-302a-40c5-886c-8515a0306b90`).  But the problem is that it looks like you are putting the values into the query directly when you should be putting variables into the query and putting the values into a map. You would add something like `"P.projectID IN (:" + DBConstants.Project.PROJECT_ID + ")"` to the query and then put (PROJECT_ID, accessibleProjectIds) into a `queryParams` map, then call the `select(query, limit, offset, queryParams)` repository method.

Reply:
@machristie Got it. Thank you!